src/zope/site/site.py

```diff
@@ -247,11 +247,15 @@ def changeSiteConfigurationAfterMove(site, event):
 
     Subscriber to :class:`~.ISite` objects in a :class:`~.IObjectMovedEvent`.
     """
+    next = _findNextSiteManager(site)
+    local_sm = site.getSiteManager()
     if event.newParent is not None:
-        next = _findNextSiteManager(site)
         if next is None:
             next = zope.component.getGlobalSiteManager()
-        site.getSiteManager().__bases__ = (next, )
+        local_sm.__bases__ = (next, )
+    else:
+        if next is not None:
+            next.removeSub(local_sm)
```

Comment:
```suggestion
    else:
        local_sm.__bases__ = ()
```

I wonder if we should just be changing the bases? This makes it more like the other case, which should help keep things in sync. The effect of removing the sub should be the same.

Reply:
Thank you, I implemented this suggestion in 4815f6f.